### PR TITLE
Use "pack file" instead of "data file"

### LINKED ISSUE #2877: Pack(File) or DataFile?
What are you trying to do?
--------------------------

When working on implementations, I always get a bit confused whether to use the name "pack"/"pack file" or "data file".

In the users documentation part "Terminology", it is referred as "Packs", but the dir in the repository layout is `/data`.

Also, in the implementations, the constant is called `DataFile` which may be mixed up with `DataBlob` but I've seen may places where variables/comments or user outputs refer to "packs".

I suggest to:

- change `DataFile` into `PackFile`
- look for "data file" et al in code/comments and change if necessary
- add an explanation in `resign.rst` that packs are stored under `/data` for historic reasons

Is this a common view? I can prepare an PR if we agree that this clarifies things.

### PR DESCRIPTION
**Pull request by aawsome**
What does this PR change? What problem does it solve?
-----------------------------------------------------

Clarify code and documentation by using "pack file" or "pack" everywhere. In some places "data file" was used. This was a bit irritating as there are also data and tree blobs and pack which contain only data blobs or only tree blobs, respectively.

Was the change discussed in an issue or in the forum before?
------------------------------------------------------------

closes #2877 

Checklist
---------

- [x] I have read the [Contribution Guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches)
- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- [x] I have *not* added tests for all changes in this PR, but tests are changed accordingly
- [x] I have *not* added documentation for the changes (in the manual) but changed the docu accordingly
- [ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (template [here](https://github.com/restic/restic/blob/master/changelog/TEMPLATE))
- [x] I have run `gofmt` on the code in all commits
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits)
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

**Comment by aawsome** [comment]
The change is also needed within #2718 - I used "data files" there and were wondering if that was confusing...

**Comment by rawtaz** [comment]
I wonder if there are any or many PRs that will have to be updated if we merge this one first? If so, it might be best to put this on ice until after merging the other ones.

**Comment by aawsome** [comment]
I assume that this doesn't affect many PRs directly, but may produce easy-to-solve merge conflicts if "nearby" lines are changed. From my PRs - as far as I can see - only #2718 and #2543 should be affected.

I would argue that this PR could be used immediately as potential merge conflicts of other PRs will not arise or be easy to solve.
On the other hand, it won't be much to fix for this PR if this really doesn't interfere with other PRs - and if there are merge conflicts, they will be also easy to fix here. When postponing this, we should just make sure that all tests still pass and we do not forget a "data file" sneaked in by another PR :wink:

**Review by MichaelEischer — APPROVED** [review]
LGTM. I've thought a bit about whether to merge this now or later on. Besides a few comment and doc changes everything else will lead to a compiler error if the name change is lost during a merge. The few merge conflicts that could arise should also be rather simple to resolve. So I think it's preferential to merge this PR just now rather than wait.

**Pull request merged**
